harness/determined/pytorch/_lr_scheduler.py

```diff
@@ -46,76 +51,31 @@ def __init__(self, scheduler: torch.optim.lr_scheduler._LRScheduler, step_mode:
                    It is up to the user to decide when to call scheduler.step(),
                    and whether to pass any arguments.
         """
-
         check.check_not_none(scheduler)
         check.check_isinstance(step_mode, LRScheduler.StepMode)
 
-        self.scheduler = scheduler
-        self.step_mode = step_mode
-
-    def step(self, *args: typing.Any, **kwargs: typing.Any) -> None:
-        """Call step() on the wrapped LRScheduler instance."""
-        check.check_eq(
-            self.step_mode,
-            LRScheduler.StepMode.MANUAL_STEP,
-            "Please use the MANUAL_STEP step mode to call step() on the scheduler.",
+        self._scheduler = scheduler
+        self._step_mode = step_mode
+        self._step_count = self._scheduler._step_count  # type: ignore
+
+    def step(self, *args: Any, **kwargs: Any) -> None:
+        check.eq(
+            self._scheduler._step_count,  # type: ignore
+            self._step_count,
+            "If you have configured Determined to manage the learning rate scheduler "
+            "step, i.e. not using MANUAL_STEP mode, you cannot call step() of the "
+            "original LR scheduler.",
         )
-        return self.scheduler.step(*args, **kwargs)
 
-    def get_last_lr(self) -> typing.List:
-        """Return last computed learning rate by current scheduler.
+        self._step_count += 1
```

Comment:
non-blocking: I'm ok leaving this check in for now, while we have the `create_lr_scheduler()` API, but in the future state where users are calling `self.context.lr_scheduler()` or whatever we end up naming it, I'd just as soon we leave this check out entirely, since it is introspecting the sub-scheduler's private members.